### PR TITLE
setup project with templates

### LINKED ISSUE #6: CLI option for setting up template for a year
Could be cool to setup template for the year with for example `cargo aoc setup`. This could generate a folder structure with some template code for task one and task two, and just have `todo!()` inside them.

Maybe we could prompt for an optional token so that we can do #3

### PR DESCRIPTION
**Pull request by seblj**
Closes #6 
Setup project structure on `cargo aoc setup`.

By default it will use the current year, but it is possible to pass `-y` or `--year` as an argument to specify a different year

### CODE REVIEW

**Comment by seblj** [comment]
I guess we don't really need the main.rs that is generated. Should I just delete it after generating the project structure with `cargo new`?

File: src/setup.rs [hunk 1]
```diff
@@ -1,4 +1,54 @@
-pub fn setup()
+use clap::ArgMatches;
+
+async fn setup_template_project(year: i32) -> Result<(), Box<dyn std::error::Error>>
 {
-    println!("Setting up structure...");
+    tokio::process::Command::new("cargo")
+        .args(&["new", &format!("year_{}", year)])
```
Comment (sivertjoe) [inline src/setup.rs:6]: Yeah, just remove the main.rs file
otherwise LGTM
```rs
tokio::fs::remove_file(format!("year_{}/main.rs")).await.unwrap();
```

**Review by sivertjoe — CHANGES_REQUESTED** [review]
Yeah, just remove

**Review by sivertjoe — COMMENTED** [review]
Very nice.
last thing, can we use [this](https://github.com/sivertjoe/Advent-of-Code/blob/master/2015/templates/main.rs) template instead?
It rearranges some stuff

**Comment by seblj** [comment]
> Very nice.
> last thing, can we use [this](https://github.com/sivertjoe/Advent-of-Code/blob/master/2015/templates/main.rs) template instead?
> It rearranges some stuff

Yes that looks better! We would probably have to change the way to get input file, but that should be the job of #3

**Review by sivertjoe — CHANGES_REQUESTED** [review]
Nice, just swap the template and we're good

**Review by sivertjoe — APPROVED** [review]

**Pull request merged**